test/free_algebra.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+
+import linear_algebra.exterior_algebra
+import linear_algebra.clifford_algebra
+
+/-!
+Tests that the ring instances for `free_algebra` and derived quotient types actually work.
+
+There is some discussion about this in
+https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/algebra.2Esemiring_to_ring.20breaks.20semimodule.20typeclass.20lookup/near/212580241
+
+In essence, the use of `attribute [irreducible] the_type` was breaking instance resolution on that
```

Comment:
I haven't been paying close attention to this issue for several weeks. Am I right in thinking this is the same issue that Reid demonstrated appeared to be due to a unification failure on account of Lean's inability to spot that two terms were definitionally equal?

If that is the case, I think it's not really accurate to say that it was instance resolution that was breaking.

I think it might also be worth adding a sentence admitting that (sadly) we never got to the bottom of this.

Reply:
I think it was both - instance resolution was finding a diamond, and lean was unable to unify the two instances as being definitionally equal. I'd be happy to accept any phrasing here, as long as:

* It includes the zulip link for context
* It indicates that `attribute [irreducible] free_algebra` is deliberately absent